test/fuzz/utility.h

```diff
@@ -37,8 +37,11 @@ inline TestRequestInfo fromRequestInfo(const test::fuzz::RequestInfo& request_in
   TestRequestInfo test_request_info;
   test_request_info.metadata_ = request_info.dynamic_metadata();
   // libc++ clocks don't track at nanosecond on OS X.
-  test_request_info.start_time_ =
-      SystemTime(std::chrono::microseconds(request_info.start_time() / 1000));
+  const auto start_time =
+      std::numeric_limits<std::chrono::seconds::rep>::max() < request_info.start_time()
```

Comment:
Thanks for biting this one off.

I think there's some unit mixing here; `start_time` is actually intended to be the time in nanoseconds, that we divide below by 1000 to get micros. Here we are comparing to seconds?

FWIW, I think the long term fix for this is to make it a Duration and fix https://github.com/lyft/protoc-gen-validate/issues/97, but that requires us to be able to bump to the latest PGV, which itself has other blockers. So, your approach here is good as a fuzz unblocker.

Reply:
> I think there's some unit mixing here; start_time is actually intended to be the time in nanoseconds, that we divide below by 1000 to get micros. Here we are comparing to seconds?

Oops. Sorry, changed it to nanoseconds (I believe I saw it as nanoseconds) in here a430a44.

> FWIW, I think the long term fix for this is to make it a Duration and fix lyft/protoc-gen-validate#97, but that requires us to be able to bump to the latest PGV, which itself has other blockers. So, your approach here is good as a fuzz unblocker.

After reading the issue, I agree on fixing `protoc-gen-validate` for the long term. That's sounds complicated yet interesting.